Sources/DatadogObjc/DDURLSessionDelegate+objc.swift

```diff
@@ -8,4 +8,49 @@ import Foundation
 import Datadog
 
 @objc
-public class DDNSURLSessionDelegate: DDURLSessionDelegate {}
+open class DDNSURLSessionDelegate: NSObject, URLSessionTaskDelegate {
+    let swiftDelegate: DDURLSessionDelegate
+
+    @objc
+    override public init() {
+        swiftDelegate = DDURLSessionDelegate()
+    }
+
+    @objc
+    public init(additionalFirstPartyHosts: Set<String>) {
+        swiftDelegate = DDURLSessionDelegate(additionalFirstPartyHosts: additionalFirstPartyHosts)
+    }
+
+    /// NOTE: RUMM-1234
+    /// `DDNSURLSessionDelegate` imitates `DDURLSessionDelegate` class
+    /// it responds to whatever `DDURLSessionDelegate` responds to and
+    /// it forwards every method call to its private `DDURLSessionDelegate` instance.
+    /// That way, `DDNSURLSessionDelegate` inherits `DDURLSessionDelegate` functionality
+    /// without actual inheritance/superclass.
+
+    // swiftlint:disable implicitly_unwrapped_optional
+    override open func forwardingTarget(for aSelector: Selector!) -> Any? {
+        return swiftDelegate
+    }
+
+    override open class func forwardingTarget(for aSelector: Selector!) -> Any? {
+        return DDURLSessionDelegate.self
+    }
+
+    override open func responds(to aSelector: Selector!) -> Bool {
+        return swiftDelegate.responds(to: aSelector)
+    }
+
+    override open class func responds(to aSelector: Selector!) -> Bool {
+        return DDURLSessionDelegate.responds(to: aSelector)
+    }
```

Comment:
This way of forwarding calls to wrapped swift delegate doesn't work if subclassing the `DDNSURLSessionDelegate`. The `super` methods are not available as they are not explicitly defined in our delegate, so the code doesn't even compile. This can be seen when defining unit test for objc → swift calls forwarding:

<img width="1062" alt="Screenshot 2021-04-08 at 10 54 47" src="https://user-images.githubusercontent.com/2358722/113998957-d9e35e00-9859-11eb-946a-ed1f73187460.png">

<img width="1099" alt="Screenshot 2021-04-08 at 10 56 49" src="https://user-images.githubusercontent.com/2358722/113998978-de0f7b80-9859-11eb-82ec-34ef43882d15.png">